android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -100,21 +128,56 @@ class RemoteSettingsClient(
     /**
      * Reads all response for a collection found in the local storage.
      */
-    suspend fun read(): RemoteSettingsResult = withContext(Dispatchers.IO) {
-        if (!file.exists()) {
-            RemoteSettingsResult.DiskFailure(FileNotFoundException("File not found"))
-        } else {
+    suspend fun read(): RemoteSettingsResult =
+        withContext(Dispatchers.IO) {
             try {
-                val fileReader = FileReader(file)
-                val jsonObject = JSONObject(fileReader.readText())
-                val response = jsonObject.get("response") as RemoteSettingsResponse
+                val fileContent = file.readText()
+                println("file $fileContent")
 
-                fileReader.close()
+                val response = parseJsonToResponse(fileContent)
                 RemoteSettingsResult.Success(response)
             } catch (e: IOException) {
+                val str = e.message
+                println("catch $str")
+
                 RemoteSettingsResult.DiskFailure(e)
             }
         }
+
+    fun parseJsonToResponse(jsonString: String): RemoteSettingsResponse {
```

Comment:
I'm not sure we want these functions to remain public. I would suggest considering this from the perspective of the consumer. Will we be expecting consumers to have JSON that they want to parse to a `RemoteSettingsResponse`?

Reply:
Sorry for all the nits. This was never meant to be the final version. I just threw around two ways in two commits if we had to manually go through this route .